SimpleTimer.cs

```diff
@@ -0,0 +1,65 @@
+using System.Collections;
+using SubsurfaceStudios.Attributes;
+using SubsurfaceStudios.Utilities.Async;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace SubsurfaceStudios.Utilities.Time
+{
+    [System.Serializable]
+    public class SimpleTimer
+    {
+        public float duration;
+        
+        [Header("Runtime")]
+        [InspectorReadOnly] public bool isRunning = false;
+        [InspectorReadOnly] public float currentTime = 0f;
+        
+        public UnityEvent<float> onTimerStart;
+        public UnityEvent<float> onTimerTick;
+        public UnityEvent<float> onTimerStop;
+
+        public SimpleTimer(float duration)
+        {
+            this.duration = duration;
+        }
+        
+        public void Start()
+        {
+            if (isRunning)
+            {
+                Debug.LogError("Timer already running!");
+                return;
+            }
+            
+            isRunning = true;
+            onTimerStart?.Invoke(currentTime);
+
+            StaticCoroutineHandler.StartCoroutineStatic(Timer());
+        }
+
+        public IEnumerator Timer()
+        {
+            while (currentTime < duration && isRunning)
+            {
+                currentTime += Time.deltaTime;
+                onTimerTick?.Invoke(currentTime);
```

Comment:
Please wrap this invocation in a try/catch block. This way, the timer won't stop mid-way through if user code throws an exception.

For reference:
```suggestion
                try {
                     // Assume user code could throw any number of errors.
                     onTimerTick?.Invoke(currentTime);
                } catch (Exception ex) {
                     Debug.LogException(ex);
                }
```